app/models/conversation.rb

```diff
@@ -0,0 +1,16 @@
+class Conversation < ActiveRecord::Base
+  belongs_to :sender, :foreign_key => :sender_id, class_name: 'User'
+  belongs_to :recipient, :foreign_key => :recipient_id, class_name: 'User'
+
+  has_many :messages, dependent: :destroy
+
+  validates_uniqueness_of :sender_id, :scope => :recipient_id
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.